core/trino-main/src/main/java/io/trino/cost/AggregationStatsRule.java

```diff
@@ -50,36 +51,46 @@ public Pattern<AggregationNode> getPattern()
     @Override
     protected Optional<PlanNodeStatsEstimate> doCalculate(AggregationNode node, StatsProvider statsProvider, Lookup lookup, Session session, TypeProvider types, TableStatsProvider tableStatsProvider)
     {
-        if (node.getGroupingSetCount() != 1) {
+        if (node.getGroupingSetCount() != 1 || node.getStep() == INTERMEDIATE) {
             return Optional.empty();
         }
 
-        if (node.getStep() != SINGLE && node.getStep() != FINAL) {
-            return Optional.empty();
+        PlanNodeStatsEstimate estimate;
+
+        if (node.getStep() == PARTIAL) {
```

Comment:
is this intentional to exclude `INTERMEDIATE` from this condition?

Reply:
For `INTERMEDIATE` this rule will return `Optional.empty()`

Reply:
`INTERMEDIATE` is disabled by default, and it is rarely used. To add estimates, firstly we should extend tests cases. This could be done, but maybe as a follow-up to this PR?